downloader/src/main/kotlin/Downloader.kt

```diff
@@ -119,6 +113,8 @@ object Downloader {
 
         outputDirectory.apply { safeMkdirs() }
 
+        if (pkg.isMetaDataOnly) return DownloadResult(dateTime = Instant.now(), downloadDirectory = outputDirectory)
```

Comment:
I'm asking myself whether we should even care to create the `outputDirectory` here, or move this before line 114. For the hypothetical case that all packages a metadata-only packages the directory would be completely empty. But then again it's probably nicer that we can rely on the `downloadDirectory` passed as part of the `DownloadResult` to exist if there was no error.

Reply:
Ok, then were both fine with the code `as-is` IIUC.

Reply:
Yes, I just wanted to document my line of thought. If what I was mentioning were also your thoughts, do you think the rationale to do `outputDirectory.apply { safeMkdirs() }` before this line deserves a code comment?

Reply:
Yes, I've actually had quite similar thoughts about this - but I'm undecided about the comment too.